### PR TITLE
Add fb login button demo

### PR DESCRIPTION
**Pull request by Ian-Stewart-Binks**
This pull request explores what user data we can get from Facebook, and how to do it through the server (Happstack) and on the client side (JavaScript).

The Scripts.hs links have been changed to be absolute, however there is most likely a better way to achieve this. This is only meant as a temporary fix, I will create an issue for this.

Note that the secret must be inserted to successfully connect to Facebook. Only David and I have access to the secret right now.

This is built on top of #269.

### CODE REVIEW

File: hs/Scripts.hs [hunk 1]
```diff
@@ -11,39 +11,39 @@ import MasterTemplate
...
-                                           "static/js/common/utilities/util.js",
-                                           "static/js/graph/utilities/structs.js",
-                                           "static/js/graph/utilities/util.js",
-                                           "static/js/graph/create_data.js",
-                                           "static/js/graph/parse_graph.js",
-                                           "static/js/graph/mouse_events.js",
-                                           "static/js/graph/setup.js"])
+                                            "http://code.jquery.com/ui/1.10.4/jquery-ui.js",
+                                            "http://localhost:8000/static/js/graph/modal.js",
+                                            "http://localhost:8000/static/js/graph/objects/edge.js",
```
Comment (david-yz-liu) [inline hs/Scripts.hs]: Why did you change these?

File: hs/server.hs [hunk 2]
```diff
@@ -32,9 +34,24 @@ about = "about"
 static :: String
 static = "static"
 
+app :: FB.Credentials
+app = FB.Credentials "localhost" "442286309258193" "INSERT_SECRET"
```
Comment (david-yz-liu) [inline hs/server.hs]: It would be helpful to add a comment here.

File: hs/Scripts.hs (hunk 1 above)
Comment (Ian-Stewart-Binks) [inline hs/Scripts.hs]: A temporary fix for #277.

File: hs/server.hs (hunk 2 above)
Comment (Ian-Stewart-Binks) [inline hs/server.hs]: I've now added a helpful comment.

File: hs/Scripts.hs (hunk 1 above)
Comment (david-yz-liu) [inline hs/Scripts.hs]: Hmmm okay... I'll ask you about this later.

Comment (Ian-Stewart-Binks) [inline hs/Scripts.hs]: I can give you a demonstration tomorrow, it shouldn't take more than a minute.

Comment (Ian-Stewart-Binks) [inline hs/Scripts.hs]: Actually, even easier: Run the server on your happstack branch, and go to `http://localhost:8000/graph/`- there should be a surprise waiting there.
